eddy/src/transcript.rs

```diff
@@ -0,0 +1,45 @@
+use crate::{limb, PublicKeyShare};
+
+pub trait TranscriptProtocol {
+    fn begin_decryption(&mut self);
+    fn begin_limb_decryption(&mut self);
+    fn append_public_key_share(&mut self, share: &PublicKeyShare);
+    fn append_limb_ciphertext(&mut self, ciphertext: &limb::Ciphertext);
+    fn append_decryption_share_point(&mut self, point: &decaf377::Element);
+    fn append_blinding_commitment(&mut self, label: &'static [u8], point: &decaf377::Element);
+
+    fn challenge_scalar(&mut self, label: &'static [u8]) -> decaf377::Fr;
+}
+
+impl TranscriptProtocol for merlin::Transcript {
+    fn begin_decryption(&mut self) {
+        self.append_message(b"dom-sep", b"eddy-decaf377-decrypt");
+    }
+    fn begin_limb_decryption(&mut self) {
+        self.append_message(b"dom-sep", b"begin-limb");
+    }
+    fn append_public_key_share(&mut self, share: &PublicKeyShare) {
+        self.append_message(b"dom-sep", b"public-key-share");
+        self.append_message(b"index", &share.participant_index.to_le_bytes());
+        self.append_message(b"public-key-share", &share.pub_key_share.compress().0);
+    }
+    fn append_limb_ciphertext(&mut self, ciphertext: &limb::Ciphertext) {
+        self.append_message(b"dom-sep", b"limb-ciphertext");
+        self.append_message(b"c1", &ciphertext.c1.compress().0);
+        self.append_message(b"c2", &ciphertext.c2.compress().0);
+    }
+    fn append_decryption_share_point(&mut self, point: &decaf377::Element) {
+        self.append_message(b"decryption-share-point", &point.compress().0);
+    }
+    fn append_blinding_commitment(&mut self, label: &'static [u8], point: &decaf377::Element) {
+        self.append_message(b"dom-sep", label);
+        self.append_message(b"blinding-commitment", &point.compress().0);
+    }
```

Comment:
While `decaf377` gives us a prime-order group, the group still does have one small subgroup -- the identity element is a subgroup of size 1, and so I think we need to check that we're not processing a bunch of identity points.

One way we could do this is suggested in the [Merlin docs](https://merlin.cool/use/protocol.html#validation-logic) -- include validation in the transcript functions, making them return `Result`s and fail on identity points.